synapse/handlers/auth.py

```diff
@@ -106,17 +132,35 @@ def __init__(self, hs):
                     if t not in login_types:
                         login_types.append(t)
         self._supported_login_types = login_types
+        # Login types and UI Auth types have a heavy overlap, but are not
+        # necessarily identical. Login types have SSO (and other login types)
+        # added in the rest layer, see synapse.rest.client.v1.login.LoginRestServerlet.on_GET.
+        ui_auth_types = login_types.copy()
+        if self._saml2_enabled:
+            ui_auth_types.append(LoginType.SSO)
+        self._supported_ui_auth_types = ui_auth_types
 
         # Ratelimiter for failed auth during UIA. Uses same ratelimit config
         # as per `rc_login.failed_attempts`.
         self._failed_uia_attempts_ratelimiter = Ratelimiter()
 
         self._clock = self.hs.get_clock()
 
-        # Load the SSO redirect confirmation page HTML template
+        # Load the SSO HTML templates.
+
+        # The following template is shown to the user before redirecting them to
+        # the SSO login page. It notifies the user they are about to be
+        # redirected and that the Synapse server will be gaining access to their
+        # SSO account.
```

Comment:
the concern is less about the synapse server getting access to their SSO account (the SSO provider will tell them that), so much as the client getting access to their matrix account.

Also: it's shown *after* the SSO login page, before redirecting back to the client.

```suggestion
        # The following template is shown to the user during a client login via SSO,
        # after the SSO completes and before redirecting them back to their client.
        # It notifies the user they are about to give access to their matrix account
        # to the client.
```